server/src/updaters/relationship-updaters.js

```diff
@@ -199,7 +201,65 @@ async function updateRelationships(
 }
 
 function sortIDs(firstId: string, secondId: string) {
-  return [firstId, secondId].map(Number).sort((a, b) => a - b);
+  return [Number(firstId), Number(secondId)].sort((a, b) => a - b);
 }
 
-export { updateRelationships };
+async function knowOfRelationshipsForThread(
+  newMemberIDs: string[],
+  threadID?: string,
+): Promise<RelationshipChangeset> {
+  const combinations = combinations2(newMemberIDs).map(
+    ([firstId, secondId]) => {
+      const [user1, user2] = sortIDs(firstId, secondId);
+      return { user1, user2, status: undirectedStatus.KNOW_OF };
+    },
+  );
+
+  if (threadID) {
+    const query = SQL`
+      SELECT user FROM memberships WHERE thread = ${threadID} ORDER BY user
```

Comment:
Nit: I always forget which direction is the default for `ORDER BY`, so I like to state it explicitly: `ORDER BY user ASC`